### PR TITLE
nullcheck to avoid 500 if asset does not exist #1414

### LINKED ISSUE #1414: Image Core component throws NPE if fileReference points to a non existing asset
https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.15.0/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java#L875

## Bug Report

**Current Behavior**
Requesting any core/wcm/components/image/v2/image component that has a fileReference which points to a nonexisting image throws a 500. This happens due to the fact that https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.15.0/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java#L875 can be null 

**Expected behavior/code**
Requesting a core/wcm/components/image/v2/image should never throw an NPE internally due to an invalid fileReference. https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.15.0/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java#L875 take invalid fileReference into account

**Environment**
- AEM 6.5 SP7 )
- Core Components version: 2.15.0

### PR DESCRIPTION
**Pull request by Noel-Voelin**
As described in the issue, requesting a core image component that has a file reference which points to a non existinga asset triggers several exceptions and results the request to be responded as a 500 instead of a 404.

I fixed this by simply null checking the imageResource before setting the source to Source.ASSET. The source will remain to be set to Source.NONEXISTING and the therefore correctly be treated as nonexisting (404)

There is a description on how to reproduce this as well as an example har file attached to the ticket. 

I added an unit-test to ensure the HttpServletResponse code to be correct.

Fixes #1414

### CODE REVIEW

**Comment by Noel-Voelin** [comment]
[Adobe CLA_encrypted_.pdf](https://github.com/adobe/aem-core-wcm-components/files/6078631/Adobe.CLA_encrypted_.pdf)

**Review by msagolj — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java [hunk 1]
```diff
@@ -873,7 +873,9 @@ private long getRequestLastModifiedSuffix(@Nullable String suffix) {
             String fileReference = component.getValueMap().get(DownloadResource.PN_REFERENCE, String.class);
             if (StringUtils.isNotEmpty(fileReference)) {
                 imageResource = component.getResourceResolver().getResource(fileReference);
-                source = Source.ASSET;
+                if(imageResource != null) {
```
Comment (vladbailescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java:876]: Please add a space between `if` and parentheses!

**Review by vladbailescu — APPROVED** [review]
Thanks for the fix! LGTM except for a small code style issue.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java (hunk 1 above)
Comment (Noel-Voelin) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java:876]:
```suggestion
                if (imageResource != null) {
```

**Pull request merged**